### PR TITLE
Refactor UI modules

### LINKED ISSUE #7704: clean up base_ui.py, extract page classes
We need to remove classes not related to base UI functionality from the base_ui.py this way we would not import any child classes to classes descending from BaseUI receiving circular import problem such as:

ImportError while loading conftest '/Users/danielosypenko/Work/automation_4/ocs-ci/tests/conftest.py'.
tests/conftest.py:28: in <module>
    from ocs_ci.ocs import constants, defaults, fio_artefacts, node, ocp, platform_nodes
ocs_ci/ocs/platform_nodes.py:15: in <module>
    from ocs_ci.deployment.vmware import (
ocs_ci/deployment/vmware.py:77: in <module>
    from .deployment import Deployment
ocs_ci/deployment/deployment.py:122: in <module>
    from ocs_ci.ocs.ui.helpers_ui import ui_deployment_conditions
ocs_ci/ocs/ui/helpers_ui.py:8: in <module>
    from ocs_ci.ocs.ui.base_ui import login_ui, close_browser
ocs_ci/ocs/ui/base_ui.py:16: in <module>
    from ocs_ci.ocs.ui.create_resource import CreateResourceForm
ocs_ci/ocs/ui/create_resource.py:12: in <module>
    from ocs_ci.ocs.ui.base_ui import PageNavigator
E   ImportError: cannot import name 'PageNavigator' from partially initialized module 'ocs_ci.ocs.ui.base_ui' (most likely due to a circular import) (/Users/danielosypenko/Work/automation_4/ocs-ci/ocs_ci/ocs/ui/base_ui.py)

### PR DESCRIPTION
**Pull request by DanielOsypenko**
FIXES #7704

Goals: 
- resolve circular import problem
- clean up base_ui leaving to it basic UI functionality

### CODE REVIEW

**Comment by DanielOsypenko** [comment]
@am-agrawa @OdedViner this change was requested by you when reviewing [odf-topology-tests ](https://github.com/red-hat-storage/ocs-ci/pull/7782) Ready for review

**Comment by DanielOsypenko** [comment]
@ebondare 
this is a test run https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/26664/consoleFull

we have some failures on 4.14 but none of them are related to current PR.
analysis of the test run https://docs.google.com/spreadsheets/d/1Ak_PbqgRiR_1ymZNd3qQY_5vjCXG8lY-AebZL9JoC9M/edit?usp=sharing

**Review by ebondare — APPROVED** [review]
LGTM

File: ocs_ci/ocs/ui/pvc_ui.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import logging
 import time
 
-from ocs_ci.ocs.ui.base_ui import PageNavigator
+from ocs_ci.ocs.ui.page_objects.page_navigator import PageNavigator
```
Comment (OdedViner) [inline ocs_ci/ocs/ui/pvc_ui.py:4]: Why did you move PageNavigator class from base_ui file?

Comment (DanielOsypenko) [reply ocs_ci/ocs/ui/pvc_ui.py:4]: This is a module of the page and it has navigation methods, same as any DataFoundationTab page or object, no difference. 
It does not have basic ui functions such as do_click(), get_element_attribute(), take_screenshot(), etc related to base_ui functionality. 
Plus it seriously reduce complexity of the base_ui.py module under `ocs_ci/ocs/ui/page_objects` directory. 
In other words every "page object" including PageNavigator now has it's own module, functionality may continue grow and we always will have full order.

**Review by OdedViner — APPROVED** [review]

**Review by sagihirshfeld — APPROVED** [review]

File: ocs_ci/ocs/ui/page_objects/bucket_class_tab.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        This function generates a random IPv4 address and a random string that is not an IP address.
+
+        Args:
+            rule_exp (str): the rule requested to be checked. rule_exp text should match the text from validation popup
+
+        Returns:
+            bool: True if the rule is satisfied for the random string that is not an IP address, False otherwise.
+        """
+
+        def _generate_ipv4_address_str():
```
Comment (ebenahar) [inline ocs_ci/ocs/ui/page_objects/bucket_class_tab.py:74]: not very crucial but maybe in a follow up PR - this doesn't seem to belong to UI functionalities. Suggest moving it to a general utils file

**Review by petr-balogh — APPROVED** [review]
